pkg/remotecmd/kubeexec.go

```diff
@@ -0,0 +1,192 @@
+		cmd = append(cmd, fmt.Sprintf("%s && cd %s && %s",
+			pidWriterCmd, devfileCmd.Exec.WorkingDir, cmdLine))
+	} else {
+		// Full command is: /bin/sh -c "echo $$ > $pidFile && $cmdLine"
+		// We are deleting the PID file if the main command does not succeed, so its status could be detected accordingly.
+		cmd = append(cmd, fmt.Sprintf("%s && %s", pidWriterCmd, cmdLine))
+	}
+
+	go func() {
+		stdout, stderr, err := Execute(kclient, podName, containerName, false, cmd...)
+
+		if err != nil {
+			klog.V(2).Infof("error while running background command: %v", err)
+
+			var codeExitError exec.CodeExitError
+			if errors.As(err, &codeExitError) && codeExitError.ExitStatus() == _sigtermExitCode {
+				//process killed by SIGTERM signal (potentially via the stop command, upon a restart command)
+				//=> displaying the logs might feel disturbing to the end-users
+				klog.V(2).Infof("stdout: %s", strings.Join(stdout, "\n"))
+				klog.V(2).Infof("stderr: %s", strings.Join(stderr, "\n"))
+			} else {
+				// Use GetStderr in order to make sure that colour output is correct
+				// on non-TTY terminals
+				log.Warningf("Last %d lines of log:", _numberOfLinesToOutputLog)
+				_ = util.DisplayLogLines(stdout, log.GetStderr(), _numberOfLinesToOutputLog)
+				_ = util.DisplayLogLines(stderr, log.GetStderr(), _numberOfLinesToOutputLog)
+			}
+		}
+	}()
+
+	return nil, nil, nil
+}
+
+func (k *kubeExecProcessHandler) StopProcessForCommand(
+	devfileCmd devfilev1.Command,
+	kclient kclient.ClientInterface,
+	podName string,
+	containerName string,
+) (stdout []string, stderr []string, err error) {
+	return Execute(kclient, podName, containerName, false,
+		common.ShellExecutable, "-c", fmt.Sprintf("(kill $(cat %[1]s) || true) && rm -f %[1]s", getPidFileForCommand(devfileCmd)))
+}
+
+func getRemoteProcessPID(kclient kclient.ClientInterface, devfileCmd devfilev1.Command, podName string, containerName string) (int, error) {
+	pidFile := getPidFileForCommand(devfileCmd)
+	stdout, stderr, err := Execute(kclient, podName, containerName, false,
+		common.ShellExecutable, "-c", fmt.Sprintf("cat %s || true", pidFile))
+
+	if err != nil {
+		return 0, err
+	}
+
+	if len(stdout) == 0 {
+		//The file does not exist. We assume the process has not run yet.
+		return 0, nil
+	}
+	if len(stdout) != 1 {
+		return 0, fmt.Errorf(
+			"unable to determine command status, due to unexpected number of lines for %s, output: %v %v",
+			pidFile, stdout, stderr)
+	}
+
+	line := stdout[0]
+	var pid int
+	pid, err = strconv.Atoi(strings.TrimSpace(line))
+	if err != nil {
+		klog.V(2).Infof("error while trying to retrieve status of command: %+v", err)
+		return 0, fmt.Errorf("unable to determine command status, due to unexpected content for %s: %s",
+			pidFile, line)
+	}
+	return pid, nil
+}
+
+// getPidFileForCommand returns the path to the PID file in the remote container.
+// The parent folder is supposed to be existing.
```

Comment:
Not figured out so far. But it's working fine, so not going to get crazy over it. 😄

Reply:
Actually, your comment made me think that instead of hardcoding the `/opt/odo` parent path here, I would use the constant already defined in the storage package :)

Reply:
Yes, I remember coming across a constant with that path value. :)